network-libp2p/src/dispatch/codecs/typed.rs

```diff
@@ -193,6 +193,55 @@ impl MessageCodec {
 
         Ok(())
     }
+
+    fn decode_serialized_message(
+        &mut self,
+        src: &mut BytesMut,
+    ) -> Result<(MessageType, BytesMut), Error> {
+        // Make a cursor, so we later know how many bytes we read
+        let mut c = Cursor::new(&src);
+
+        let header = Header::deserialize(&mut c)?;
+        // Deserializing the header was successful
+        let header_length = c.position() as usize;
+
+        // Drop the cursor, so we can mut-borrow the `src` buffer again.
+        drop(c);
+
+        // Preliminary header check (we can't verify the checksum yet)
+        header.preliminary_check()?;
+
+        // Now parse the body.
+        if src.len() >= header.length as usize {
+            // We have read enough bytes to read the full message
```

Comment:
This is kinda misleading, it sounds like it could wait for more data.
```suggestion
            // We have at least as much bytes as the header indicated
```

Could this be an equality check instead?

Reply:
> Could this be an equality check instead?

You're right. It should be an equality.